src/iidm/converter/xml/VoltageLevelXml.cpp

```diff
@@ -48,6 +56,58 @@ const char* VoltageLevelXml::getRootElementName() const {
     return VOLTAGE_LEVEL;
 }
 
+void VoltageLevelXml::readBusBreakerTopology(VoltageLevel& voltageLevel, NetworkXmlReaderContext& context) const {
+    context.getReader().readUntilEndElement(BUS_BREAKER_TOPOLOGY, [&voltageLevel, &context]() {
+        if (context.getReader().getLocalName() == BUS) {
+            BusXml::getInstance().read(voltageLevel, context);
+        } else if (context.getReader().getLocalName() == SWITCH) {
+            BusBreakerViewSwitchXml::getInstance().read(voltageLevel, context);
+        } else {
+            throw AssertionError(logging::format("Unsupported element %1%", context.getReader().getLocalName()));
+        }
+    });
+}
+
+void VoltageLevelXml::readCalculatedBus(VoltageLevel &voltageLevel, NetworkXmlReaderContext& context) const {
+    IidmXmlUtil::assertMinimumVersion(VOLTAGE_LEVEL, BUS, xml::ErrorMessage::NOT_SUPPORTED, xml::IidmXmlVersion::V1_1(), context.getVersion());
+    double v = context.getReader().getOptionalAttributeValue(V, stdcxx::nan());
+    double angle = context.getReader().getOptionalAttributeValue(ANGLE, stdcxx::nan());
+    std::shared_ptr<std::string> nodesString = std::make_shared<std::string>(context.getReader().getAttributeValue(NODES));
```

Comment:
I though we agree to pass the nodeString (rename to strNodes) as a copy, without using a shared_ptr.